### PR TITLE
Added an explanation on manipulating datasets

### PR DESCRIPTION
**Pull request by Helveg**
As discussed in https://github.com/h5py/h5py/pull/1328

### CODE REVIEW

**Comment by takluyver** [comment]
Thanks! I think this is substantially overlapping with the *Reading and writing data* section further down the page. We should enhance that and move it further up the page (before all the discussion of how HDF5 actually stores data), rather than adding another section, IMO.

Do you want to have a go at this?

**Comment by Helveg** [comment]
Yea no, you're right, the "Reading and writing" section is exactly what I was looking for. It's just such basic information and what comes before it is so technical and specific that you assume it's not there. I'm not familiar with the syntax used in the docs but I think that if you just move up that section the document will be more clear

**Comment by takluyver** [comment]
:+1: do you want to move the section up? The CI should check that the docs still build OK, though it's slow at the moment because we're doing a sprint (Hello from Grenoble!), so there are lots of changes happening.

If not, we'll leave this PR open as a reminder and someone here can work on the docs.

**Comment by Helveg** [comment]
Sure!

**Comment by Helveg** [comment]
@takluyver How do you make links to other sections with this doc format? I'd like to add a reference to creating empty datasets from "creating datasets" and to "fancy indexing" from "reading and writing data" :)

File: docs/high/dataset.rst [hunk 1]
```diff
@@ -303,18 +314,6 @@ list of points to select, so be careful when using it with large masks::
    Selecting using an empty list is now allowed.
    This returns an array with length 0 in the relevant dimension.
 
-.. _dataset_iter:
```
Comment (takluyver) [inline docs/high/dataset.rst]: Can you preserve this line above the header? This makes a label for cross-linking.

To make internal links to a label, you'd write:

```rst
:ref:`dataset_iter` (using the text from the heading)
:ref:`iterating over datasets <dataset_iter>` (overriding it with custom text)
```

**Comment by Helveg** [comment]
Ok I can get to preserve that, but it gave an error in the build of docs. I
changed the line below it from a section to a subsection and there were no
other references to it.

How would you add that line and avoid the error? I think the error said
that it required a caption (?) if it wasn't followed by a section header.

File: docs/high/dataset.rst (hunk 1 above)
Comment (Helveg) [reply docs/high/dataset.rst]: Ok I can try to preserve that, but it gave an error in the build of docs. I changed the line below it from a section to a subsection and there were no other references to it. How would you add that line and avoid the error? I think the error said that it required a caption (?) if it wasn't followed by a section header.

Comment (takluyver) [reply docs/high/dataset.rst]: It should be OK if it's just above the header, like the other ones. The heading level (underline style) shouldn't matter.

You can try building the docs locally with `make html` if you have a local checkout. Or I can try to have a look at some point.

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**